### PR TITLE
feat: add certificate preference to name fields

### PR DESCRIPTION
**Pull request by bseverino**
Adds a checkbox under "Name" and "Verified Name" fields, signifying which name the user prefers to display on their
certificates. When checking the checkbox, the user can save this choice along with their name. When unchecking the check box, a modal appears, prompting the user to choose a name to display on their certificate.

---
#### Default messaging

Screen Shot 2021-09-13 at 9 46 43 AM
---
#### Full name is selected for certificates

Screen Shot 2021-09-13 at 9 46 58 AM
---
### Selecting verified name to be used on certificates instead

https://user-images.githubusercontent.com/10442143/133097478-e3b7a1d7-04a3-4a14-bcbd-8793f08c4e93.mov

---
### Attempting to deselect the current choice brings up a modal

https://user-images.githubusercontent.com/10442143/133097544-cb34156e-413f-4615-aa36-467bac28536c.mov

### CODE REVIEW

File: src/account-settings/certificate-preference/CertificatePreference.jsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+function CertificatePreference({
+  intl,
+  fieldName,
+  originalFullName,
+  originalVerifiedName,
+  saveState,
+  useVerifiedNameForCerts,
+}) {
+  if (!originalVerifiedName) {
```
Comment (alangsto) [inline src/account-settings/certificate-preference/CertificatePreference.jsx:32]: I may have missed this elsewhere, but is there an additional check to only display this component if the verified name feature is enabled?

Comment (bseverino) [reply src/account-settings/certificate-preference/CertificatePreference.jsx:32]: Nope good catch, this is probably the best place to add it

File: src/account-settings/EditableField.jsx [hunk 2]
```diff
@@ -103,54 +104,57 @@ function EditableField(props) {
...
-                  // we would have to add that logic for every field given our
-                  // current structure of the application.
-                  if (saveState === 'pending') { e.preventDefault(); }
-                }}
-                disabledStates={[]}
-              />
-              <Button
-                variant="outline-primary"
-                onClick={handleCancel}
+          <>
```
Comment (schenedx) [inline src/account-settings/EditableField.jsx:107]: Why we need this empty element?

File: src/account-settings/certificate-preference/test/CertificatePreference.test.jsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+jest.mock('@edx/frontend-platform/auth');
+jest.mock('../../data/selectors', () => jest.fn().mockImplementation(() => ({ certPreferenceSelector: () => ({}) })));
+
+const history = createMemoryHistory();
+
+const IntlCertificatePreference = injectIntl(CertificatePreference);
+
+const mockStore = configureStore();
+
+describe('NameChange', () => {
```
Comment (schenedx) [inline src/account-settings/certificate-preference/test/CertificatePreference.test.jsx:36]: Love the unit tests!

**Review by schenedx — COMMENTED** [review]
I am excited of this change. No further feedback at the moment.
I can review later today once you address inline comments.

File: src/account-settings/EditableField.jsx (hunk 2 above)
Comment (bseverino) [reply src/account-settings/EditableField.jsx:107]: It's required to return multiple children from one component: https://reactjs.org/docs/fragments.html

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**